code/modules/admin/admin_fax_panel.dm

```diff
@@ -0,0 +1,134 @@
+		return
+
+	var/datum/fax_panel_interface/ui = new(usr)
+	ui.ui_interact(usr)
+
+/// Panel
+/datum/fax_panel_interface
+	/// All faxes in game list
+	var/available_faxes = list()
+	/// List with available stamps
+	var/stamp_list = list() // I need to understand, how to remove chameleon stamps...
+
+	/// Paper which admin edit and send.
+	var/obj/item/paper/fax_paper = new /obj/item/paper(null)
+
+	/// Default name of fax. Used when field with fax name not edited.
+	var/sending_fax_name = "Secret"
+	/// Default name of paper. paper - bluh-bluh. Used when field with paper name not edited.
+	var/default_paper_name = "Standart Report"
+
+/datum/fax_panel_interface/New()
+	//Get all faxes, and save them to our list.
+	for(var/obj/machinery/fax/fax in GLOB.machines)
+		available_faxes += fax
+	//Get all stamps
+	for(var/stamp in subtypesof(/obj/item/stamp))
+		var/obj/item/stamp/real_stamp = stamp 
+		if(length(initial(real_stamp.actions)) == 0)// try to remove chameleon, dont work.
+			stamp_list += list(list(initial(real_stamp.name), initial(real_stamp.icon_state)))
+	//Give our paper special status, to read everywhere.
+	fax_paper.request_state = TRUE
+
+//Maybe.. useless proc, but find fax like object what we need.
+/datum/fax_panel_interface/proc/get_fax_by_name(name)
+	if(!length(available_faxes))
+		return
+
+	for(var/obj/machinery/fax/potential_fax in available_faxes)
+		if(potential_fax.fax_name == name)
+			return potential_fax
+	return
+
+/datum/fax_panel_interface/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AdminFax")
+		ui.open()
+
+/datum/fax_panel_interface/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/fax_panel_interface/ui_static_data(mob/user)
+	var/list/data = list()
+	data["faxes"] = list()
+	data["stamps"] = list()
+	for(var/stamp in stamp_list)
+		data["stamps"] += list(stamp[1]) // send only names.
+	for(var/obj/machinery/fax/another_fax in available_faxes)
+		data["faxes"] += list(another_fax.fax_name)
+	return data
+
+/datum/fax_panel_interface/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if(..())
+		return
+
+	if(!check_rights(R_ADMIN))
+		return
+	
+	var/obj/machinery/fax/action_fax // fax what we find later
+
+	if(params["faxName"])
+		action_fax = get_fax_by_name(params["faxName"]) // that is
+	
+	switch(action)
+
+		if("jump") // jump on fax. maybe i shoul make follow action(?)
+			var/turf/fax_turf = get_turf(action_fax)
+			if(!fax_turf || !usr.client)
+				return
+			
+			usr.client.jumptoturf(fax_turf)
+		
+		if("preview") // see saved variant
+			if(!fax_paper)
+				return
+			
+			fax_paper.ui_interact(usr)
+		
+		if("save") // save paper
+			if(params["paperName"])
+				default_paper_name = params["paperName"]
+			if(params["fromWho"])
+				sending_fax_name = params["fromWho"]
+			
+			fax_paper.ui_status(usr, UI_CLOSE) // i wannd reload, or close and open UI, but i dont know how. need help
```

Comment:
I don't know what you mean by this comment? Why do you want to reload the UI?

Reply:
i wanna, just reload paper UI when user save paper info(
otherwise he needs to close the interface on his own and reopen)

Reply:
Can you just call `ui_interact`?

Reply:
What did your code look like

Reply:
on 103, just fax_paper.ui_interact(usr)